test/backward-stacktrace.h

```diff
@@ -0,0 +1,31 @@
+#include "backward.hpp"
+#include "gtest/gtest.h"
+
+namespace backward_stacktrace {
+
+// Do set-up and tear-down at the test program level.
+// When RUN_ALL_TESTS() is called, it first calls the
+// SetUp() method of the environment object, then runs
+// the tests if there was no fatal failures, and finally
+// calls TearDown() of the environment object. If there
+// was any failure - a simple helper class
+// `backward::SignalHandling` will register this for us
+// and print out consistent stacktrace.
+// (check_line_length skip)
+// https://github.com/YOU-i-Labs/googletest/blob/master/googletest/docs/V1_7_AdvancedGuide.md#global-set-up-and-tear-down
+class BackwardStackTrace : public ::testing::Environment {
+ public:
+  ~BackwardStackTrace() override {}
+
+  void SetUp() override {
+    // A simple helper class that registers for you the most
+    // common signals and other callbacks to segfault,
+    // hardware exception, un-handled exception etc.
+    // https://github.com/bombela/backward-cpp#signalhandling
+    static backward::SignalHandling sh;
```

Comment:
Please comment on why this is static.

Can it instead be a member variable of BackwardStackTrace?

The style guide prohibits static variables with non-trivial destructors: https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables if this can't be a member variable, then next best would be to make this a static pointer (that never gets deleted)